### PR TITLE
[SPARK-30322][DOCS] Add stage level scheduling docs

### PR DESCRIPTION
**Pull request by tgravescs**
### What changes were proposed in this pull request?

Document the stage level scheduling feature.

### Why are the changes needed?

Document the stage level scheduling feature.

### Does this PR introduce _any_ user-facing change?

Documentation.

### How was this patch tested?

n/a docs only

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -3028,3 +3028,10 @@ There are configurations available to request resources for the driver: <code>sp
...
+
+See the `RDD.withResources` and `ResourceProfileBuilder` api's for using this feature. The current implementation acquires new executors for each ResourceProfile created and currently has to be an exact match. Spark does not try to fit tasks into an executor that require a different ResourceProfile than the executor was created with. Executors that are not in use will idle timeout with the dynamic allocation logic. The default configuration for this feature is to only allow one ResourceProfile per stage. If the user associates more then 1 ResourceProfile to an RDD, Spark will throw an exception by default. See config <code>spark.scheduler.resource.profileMergeConflicts</code> to control that behavior. The current merge strategy Spark implements when <code>spark.scheduler.resource.profileMergeConflicts</code> is enabled is a simple max of each resource within the conflicting ResourceProfiles. Spark will create a new ResourceProfile with the max of each of the resources.
```
Comment (HyukjinKwon) [inline docs/configuration.md:3037]: maybe  `<code>spark.scheduler.resource.profileMergeConflicts</code>` -> `` `spark.scheduler.resource.profileMergeConflicts` ``. I believe they are same. Also maybe `ResourceProfile` -> `` `ResourceProfile` ``.

File: docs/configuration.md [hunk 2]
```diff
@@ -3028,3 +3028,10 @@ There are configurations available to request resources for the driver: <code>sp
...
+
+# Stage Level Scheduling Overview
+
+The stage level scheduling feature allows users to specify task and executor resource requirements at the stage level. This allows for different stages to run with executors that have different resources. A prime example of this is one ETL stage runs with executors with just CPUs, the next stage is an ML stage that needs GPUs. Stage level scheduling allows for user to request different executors that have GPUs when the ML stage runs rather then having to acquire executors with GPUs at the start of the application and them be idle while the ETL stage is being run.
+This is only available for the RDD api in Scala, Java, and Python and requires dynamic allocation to be enabled.  It is only available on YARN at this time. See the [YARN](running-on-yarn.html#stage-level-scheduling-overview) page for more implementation details.
```
Comment (HyukjinKwon) [inline docs/configuration.md:3035]: nit: api -> API

File: docs/configuration.md (hunk 1 above)
Comment (tgravescs) [reply docs/configuration.md:3037]: I use ResourceProfile a lot, how about I do it to the first one and then leave the rest?

Comment (HyukjinKwon) [reply docs/configuration.md]: oh then it's alright.

**Comment by HyukjinKwon** [comment]
From my reading, looks good.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thanks, @tgravescs and @HyukjinKwon .

**Comment by tgravescs** [comment]
merged to master, thanks @dongjoon-hyun @HyukjinKwon

**Pull request closed without merging**